src/components/hoc/withAdapter.jsx

```diff
@@ -0,0 +1,71 @@
+import React, {useEffect, useState} from 'react';
+
+import WebexDataProvider from '../WebexDataProvider/WebexDataProvider';
+
+/**
+ * withAdapter is a high-order component that takes a component (WrappedComponent)
+ * and returns a new component (WithAdapter) that, given an adapter, takes care of
+ * connecting/disconnecting this adapter instance and providing in on a react context.
+ *
+ * Example:
+ *
+ * function OriginalComponent {
+ *   return <div> </div>;
+ * }
+ *
+ * createAdapter: props => {
+ *   const webex = new Webex({
+ *     credentials: props.accessToken,
+ *   });
+ *   return new WebexSDKAdapter(webex);
+ * }
+ *
+ * const EnhancedComponent = withAdapter(OriginalComponent, createAdapter);
+ *
+ * <EnhancedComponent accessToken="TOKEN"/>
```

Comment:
Let's keep the example generic
```suggestion
 * <EnhancedComponent {props} />
```